hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/infoschema/SqlInfoSchemaTest.java

```diff
@@ -110,13 +113,26 @@ public void test_tables() {
 
     @Test
     public void test_datalinks() {
+        // given
         String type = DataLinkTestUtil.DummyDataLink.class.getName();
         String wrappedType = "\"" + type + "\"";
-        sqlService.execute("CREATE DATA LINK dl TYPE " + wrappedType + " OPTIONS ()");
+
+        // create config-originated data link
+        getNodeEngineImpl(instance()).getDataLinkService().createConfigDataLink(
+                new DataLinkConfig()
+                        .setName("c_dl")
+                        .setClassName(type)
+        );
+
+        // create SQL-originated data link
+        sqlService.execute("CREATE DATA LINK s_dl TYPE " + wrappedType);
 
         assertRowsAnyOrder(
                 "SELECT * FROM information_schema.datalinks",
-                singletonList(new Row("hazelcast", "public", "dl", type, "{}"))
+                asList(
+                        new Row("hazelcast", "public", "s_dl", type, "{}", SQL),
```

Comment:
`datalink_source` column is defined as VARCHAR, we return enum without any backward/forward compatibility mechanism. I suppose we should return string instead of enum.